core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/BoxImpl.java

```diff
@@ -42,12 +42,20 @@
   @ManyToAny(metaColumn = @Column(name = "targetType"))
   @MapKeyColumn(name = "position", unique = true)
   @MapKeyClass(String.class)
-  @JoinTable(name = "BoxPosition", joinColumns = { @JoinColumn(name = "targetId") }, inverseJoinColumns = {
-      @JoinColumn(name = "boxId") })
+  @JoinTable(name = "BoxPosition", joinColumns = { @JoinColumn(name = "boxId") }, inverseJoinColumns = {
+      @JoinColumn(name = "targetId") })
   @AnyMetaDef(idType = "long", metaType = "string", metaValues = {
       @MetaValue(targetEntity = LibraryImpl.class, value = "L"),
+      @MetaValue(targetEntity = DetailedLibraryImpl.class, value = "LD"),
       @MetaValue(targetEntity = PoolImpl.class, value = "P"),
-      @MetaValue(targetEntity = SampleImpl.class, value = "S")
+      @MetaValue(targetEntity = SampleImpl.class, value = "S"),
+      @MetaValue(targetEntity = SampleAliquotImpl.class, value = "SA"),
+      @MetaValue(targetEntity = IdentityImpl.class, value = "SI"),
+      @MetaValue(targetEntity = SampleStockImpl.class, value = "SS"),
+      @MetaValue(targetEntity = SampleTissueImpl.class, value = "ST"),
+      @MetaValue(targetEntity = SampleTissueProcessingImpl.class, value = "SP"),
+      @MetaValue(targetEntity = SampleCVSlideImpl.class, value = "SC"),
+      @MetaValue(targetEntity = SampleLCMTubeImpl.class, value = "SL"),
```

Comment:
Can these be changed to something more descriptive like "Sample" and "SampleTissue" instead of "S" and "ST"?